api/pkg/s3/bucketlifecycleput.go

```diff
@@ -202,8 +218,50 @@ func (s *APIService) BucketLifecyclePut(request *restful.Request, response *rest
 
 				s3Rule.AbortIncompleteMultipartUpload = convertRuleUploadToS3Upload(rule.AbortIncompleteMultipartUpload)
 
-				// add to the s3 array
-				bucket.LifecycleConfiguration = append(bucket.LifecycleConfiguration, &s3Rule)
+				//check if the there is any modification in rule if it already exists
+				//TODO need to write another modify lifecycle API
+				//Fixme
+				lcCfg := bucket.GetLifecycleConfiguration()
+				for _, cfg := range lcCfg {
+					if cfg.Id == s3Rule.Id {
+						isModifyRequest := s.checkModifyLifecycleConfig(cfg, &s3Rule)
+						if isModifyRequest {
+
+							/*
+								set the modify flag
+							 */
+							modifyFlg = true
+
+							/*
+							Delete previous existing rule from bucket and add new rule with changes
+							 */
+							deleteLcInput := s3.DeleteLifecycleInput{Bucket: bucket.Name, RuleID: cfg.Id}
```

Comment:
ayyyoooo...why are we deleting the rule here ? what if our next update fails ?
just call the below API with the modified rules (that way, the bucket is NEVER without any rules. what if while we are doing all this, some rule timer expires and action has to be taken by S3 ?)
`PutBucketLifecycle`